substrate/frame/multi-asset-bounties/src/lib.rs

```diff
@@ -0,0 +1,1962 @@
+		/// - `asset_kind`: An indicator of the specific asset class to be funded.
+		/// - `value`: The total payment amount of this parent bounty, curator fee included.
+		/// - `curator`: Address of parent bounty curator.
+		/// - `fee`: Payment fee to parent bounty curator for execution.
+		/// - `description`: The description of this bounty.
+		///
+		/// ## Events
+		/// Emits [`Event::BountyFunded`] if successful.
+		///
+		/// ## Complexity
+		/// - O(1).
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config<I>>::WeightInfo::propose_bounty(description.len() as u32))]
+		pub fn fund_bounty(
+			origin: OriginFor<T>,
+			asset_kind: Box<T::AssetKind>,
+			#[pallet::compact] value: BalanceOf<T, I>,
+			curator: AccountIdLookupOf<T>,
+			#[pallet::compact] fee: BalanceOf<T, I>,
+			description: Vec<u8>,
+		) -> DispatchResult {
+			let max_amount = T::SpendOrigin::ensure_origin(origin)?;
+			let curator = T::Lookup::lookup(curator)?;
+			let bounded_description: BoundedVec<_, _> =
+				description.try_into().map_err(|_| Error::<T, I>::ReasonTooBig)?;
+			ensure!(fee < value, Error::<T, I>::InvalidFee);
+
+			let native_amount = T::BalanceConverter::from_asset_balance(value, *asset_kind.clone())
+				.map_err(|_| Error::<T, I>::FailedToConvertBalance)?;
+			ensure!(native_amount >= T::BountyValueMinimum::get(), Error::<T, I>::InvalidValue);
+			ensure!(native_amount <= max_amount, Error::<T, I>::InsufficientPermission);
+
+			with_context::<SpendContext<BalanceOf<T, I>>, _>(|v| {
+				let context = v.or_default();
+				let funding = context.spend_in_context.entry(max_amount).or_default();
+
+				if funding.checked_add(&native_amount).map(|s| s > max_amount).unwrap_or(true) {
+					Err(Error::<T, I>::InsufficientPermission)
+				} else {
+					*funding = funding.saturating_add(native_amount);
+					Ok(())
+				}
+			})
+			.unwrap_or(Ok(()))?;
+
+			let index = BountyCount::<T, I>::get();
+			let payment_status = Self::do_process_funding_payment(
+				index.clone(),
+				None,
+				*asset_kind.clone(),
+				value.clone(),
+				None,
+			)?;
+
+			let bounty = BountyOf::<T, I> {
+				asset_kind: *asset_kind,
+				value,
+				fee,
+				curator_deposit: 0u32.into(),
+				status: BountyStatus::FundingAttempted {
+					curator,
+					payment_status: payment_status.clone(),
+				},
+			};
+			Bounties::<T, I>::insert(index, &bounty);
+			BountyCount::<T, I>::put(index + 1);
+			BountyDescriptions::<T, I>::insert(index, bounded_description);
+
+			Self::deposit_event(Event::<T, I>::BountyFunded { index });
+
+			Ok(())
+		}
+
+		/// Fund a new child-bounty, iniitiating the payment from the parent bounty
+		/// to the child-bounty account/location.
+		///
+		/// ## Dispatch Origin
+		/// Must be signed by the parent curator.
+		///
+		/// ## Details
+		/// - If both `curator` and `fee` are not provided, the child-bounty will default to using
+		///   the parent curator and a fee of `0`. This allows the parent curator to immediately
```

Comment:
the wording is a bit surprising, curator will default to parent regardless of fee being provided or not.
Maybe better saying that curator will default to parent and fee will default to 0.